dotnet/src/webdriver/ShadowRoot.cs

```diff
@@ -34,68 +37,67 @@ public class ShadowRoot : ISearchContext, IWrapsDriver, IWebDriverObjectReferenc
+        /// <exception cref="ArgumentNullException">If <paramref name="parentDriver"/> or <paramref name="id"/> are <see langword="null"/>.</exception>
         public ShadowRoot(WebDriver parentDriver, string id)
         {
-            this.driver = parentDriver;
-            this.shadowRootId = id;
+            this.driver = parentDriver ?? throw new ArgumentNullException(nameof(parentDriver));
+            this.shadowRootId = id ?? throw new ArgumentNullException(nameof(id));
         }
 
         /// <summary>
         /// Gets the <see cref="IWebDriver"/> driving this shadow root.
         /// </summary>
-        public IWebDriver WrappedDriver
-        {
-            get { return this.driver; }
-        }
+        public IWebDriver WrappedDriver => this.driver;
 
         /// <summary>
         /// Gets the internal ID for this ShadowRoot.
         /// </summary>
-        string IWebDriverObjectReference.ObjectReferenceId
-        {
-            get { return this.shadowRootId; }
-        }
+        string IWebDriverObjectReference.ObjectReferenceId => this.shadowRootId;
 
-        internal static bool ContainsShadowRootReference(Dictionary<string, object> shadowRootDictionary)
+        internal static bool TryCreate(WebDriver parentDriver, Dictionary<string, object?> shadowRootDictionary, [NotNullWhen(true)] out ShadowRoot? shadowRoot)
         {
-            if (shadowRootDictionary == null)
+            if (shadowRootDictionary is null)
             {
                 throw new ArgumentNullException(nameof(shadowRootDictionary), "The dictionary containing the shadow root reference cannot be null");
             }
 
-            return shadowRootDictionary.ContainsKey(ShadowRootReferencePropertyName);
-        }
+            if (shadowRootDictionary.TryGetValue(ShadowRootReferencePropertyName, out object? shadowRootValue))
+            {
+                shadowRoot = new ShadowRoot(parentDriver, shadowRootValue?.ToString()!);
+                return true;
+            }
 
-        internal static ShadowRoot FromDictionary(WebDriver driver, Dictionary<string, object> shadowRootDictionary)
-        {
-            return new ShadowRoot(driver, shadowRootDictionary[ShadowRoot.ShadowRootReferencePropertyName].ToString());
+            shadowRoot = null;
+            return false;
         }
 
         /// <summary>
         /// Finds the first <see cref="IWebElement"/> using the given method.
         /// </summary>
         /// <param name="by">The locating mechanism to use.</param>
         /// <returns>The first matching <see cref="IWebElement"/> on the current context.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="by"/> is <see langword="null"/>.</exception>
         /// <exception cref="NoSuchElementException">If no element matches the criteria.</exception>
         public IWebElement FindElement(By by)
         {
-            if (by == null)
+            if (by is null)
```

Comment:
I propose to leave it for future, not related to this PR.

Reply:
I can leave it out of this PR, but passively adding modernizations as we go will prevent any "9999 files changed" PRs. In this case, the code works exactly the same (slightly optimized, since `By`'s `==` operator takes care of the null check the same way) 
![image](https://github.com/user-attachments/assets/203c3d25-4e3e-47ea-94e5-07077180882f)

Reply:
Good point! Agree.